### ISSUE TITLE
Filter not working correctly when two navigation properties have the same type.

### ISSUE THREAD

**Issue by PaulJam** [jcachat/EntityFramework.DynamicFilters#17]
Hi,

I have noticed that when an entity has two navigation properties of the same type, then the filter is only applied on the first navigation property.
A similar thing happens if a navigation property has the same type as the parent. Then the navigation property isn't filtered either.

Here is an example:
C# Console Application with the following NuGet packages:
EntityFramework version="6.1.2"
EntityFramework.DynamicFilters version="1.3.5.1"

Expected result:
The navigation properties (Nav1, Nav2, Nav3) are all null because they have IsDeleted = true.
Actual result:
Only Nav1 is null.

``` C#
using System;
using System.Data.Entity;
using System.Linq;
using EntityFramework.DynamicFilters;

namespace DynamicFilters
{
    class Program
    {
        static void Main()
        {
            var context = new Context();

            // Seed the entity.
            context.Entities.Add(new EntityA
            {
                IsDeleted = false,
                Nav1 = new EntityB
                {
                    IsDeleted = true
                },
                Nav2 = new EntityB
                {
                    IsDeleted = true
                },
                Nav3 = new EntityA
                {
                    IsDeleted = true
                }
            });
            context.SaveChanges();

            // Create a new context in order to have a clear change tracker.
            context.Dispose();
            context = new Context();

            // Get entity.
            var entity = context.Entities
                .Include(e => e.Nav1)
                .Include(e => e.Nav2)
                .Include(e => e.Nav3)
                .First();

            Console.WriteLine("Nav1: {0}\nNav2: {1}\nNav3: {2}", entity.Nav1, entity.Nav2, entity.Nav3);

            // Wait
            Console.WriteLine("End. Press key...");
            Console.ReadKey();
        }
    }

    class Context : DbContext
    {
        public IDbSet<EntityA> Entities { get; set; }

        static Context()
        {
            Database.SetInitializer(new DropCreateDatabaseAlways<Context>());
        }

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Filter("SoftDelete",
                (ISoftDelete entity) => entity.IsDeleted,
                false);

            base.OnModelCreating(modelBuilder);
        }
    }

    interface ISoftDelete
    {
        bool IsDeleted { get; set; }
    }

    class EntityA : ISoftDelete
    {
        public int Id { get; set; }
        public bool IsDeleted { get; set; }
        public EntityB Nav1 { get; set; }
        public EntityB Nav2 { get; set; }
        public EntityA Nav3 { get; set; }
    }

    class EntityB : ISoftDelete
    {
        public int Id { get; set; }
        public bool IsDeleted { get; set; }
    }
}
```

**Comment by jcachat** [comment]
Thanks for the report.  I just checked in a fix for this and updated NuGet.

**Comment by PaulJam** [comment]
Thanks!
I can confirm that it is fixed in version 1.3.6.
